dataduct/database/tests/test_database.py

```diff
@@ -0,0 +1,232 @@
+        """Database initialization with duplicate relations
+        """
+        Database(relations=[self.basic_table, self.basic_table])
+
+    def test_database_copy(self):
+        """Copying a database is a deepcopy
+        """
+        database = Database(relations=[self.basic_table])
+        database_copy = database.copy()
+
+        # Check that the copied database contains the relation
+        assert_not_equal(
+            database_copy.relation(self.basic_table.full_name), None)
+
+        # Delete the relation in the copy
+        database_copy._relations = {}
+
+        # Check that the original database still contains the relation
+        assert_not_equal(
+            database.relation(self.basic_table.full_name), None)
+
+    def test_database_has_cycles(self):
+        """Check if a database has cycles
+        """
+        database = Database(relations=[self.first_table_dependent,
+                                       self.second_table_dependent])
+        eq_(database.has_cycles(), True)
+
+    def test_database_has_no_cycles(self):
+        """Check if a database has no cycles
+        """
+        database = Database(relations=[self.first_table_dependent,
+                                       self.second_table])
+        eq_(database.has_cycles(), False)
+
+    def test_database_has_no_cycles_2(self):
+        """Check if a database has no cycles
+        """
+        database = Database(relations=[self.first_table,
+                                       self.second_table_dependent])
+        eq_(database.has_cycles(), False)
+
+    def test_database_sorted_relations(self):
+        """Get the topological sort of the database
+        """
+        database = Database(relations=[self.first_table_dependent,
+                                       self.second_table])
+        relations = database.sorted_relations()
+
+        # Verify that the relations are sorted correctly
+        eq_(len(relations), 2)
+        eq_(relations[0].table_name, self.second_table.full_name)
+        eq_(relations[1].table_name, self.first_table_dependent.full_name)
+
+    @raises(RuntimeError)
+    def test_database_sorted_relations_cyclic(self):
+        """Get the topological sort of the database with cycles
+        """
+        database = Database(relations=[self.first_table_dependent,
+                                       self.second_table_dependent])
+        database.sorted_relations()
+
+    def _test_database_scripts(self, function_name, expected_sql, **kwargs):
+        """Generate SQL scripts with a preset database
+        """
+        func = getattr(self.script_database, function_name)
+        eq_(func(**kwargs).sql(), expected_sql)
+
+    def test_database_create_relations_script(self):
+        """Creating relations in the database
+        """
+        self._test_database_scripts(
+            'create_relations_script',
+            'CREATE TABLE test_table ( id INTEGER );\n'
```

Comment:
string concatenation like this is bad, write a multiline string variable and then pass that. Same for all the tests below